### PR TITLE
Copy doc files from new to old location.

### PR DESCRIPTION
**Pull request by godofredoc**
Changes to the objectc docs to make destination paths consistent cause
failures in the documentation scripts becase some of the files could not
be found anymore. This change copies files to the old location if it
identifies the new location exist.

Bug: https://github.com/flutter/flutter/issues/109229

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by godofredoc** [comment]
This is needed to reland https://github.com/flutter/engine/pull/35265

File: dev/tools/java_and_objc_doc.dart [hunk 1]
```diff
@@ -63,10 +64,30 @@ Future<void> generateDocs(String url, String docName, String checkFile) async {
     }
   }
 
+  /// If object then copy files to old location if the archive is using the new location.
```
Comment (christopherfujino) [inline dev/tools/java_and_objc_doc.dart:67]: can you give me some context on why we have a new and old location? are the contents different?

Comment (godofredoc) [reply dev/tools/java_and_objc_doc.dart:67]: The contents are the same but the folder structure added an objectc_docs folder and puts all the contents there.

The context of that change is that with recipes v2 migration we are moving the generation of the zip files from the recipes to the engine and having the additional folder helps us prevent adding additional files to the zip if for some reason a nonempty directory is passed to the tools/gen_objcdoc.sh script.

After removing the migration to engine v2 is complete we can get rid of this additional complexity.

Comment (christopherfujino) [reply dev/tools/java_and_objc_doc.dart:67]: SGTM

**Review by christopherfujino — APPROVED** [review]
LGTM

**Pull request merged**